routes/plan.js

```diff
@@ -152,59 +186,180 @@ const getStartYear = (year) => {
+      //delete plan_id from user
+      await users
         .findByIdAndUpdate(
-          //delete plan_id from user
           plan.user_id,
           { $pull: { plan_ids: plan._id } },
           { new: true, runValidators: true }
         )
         .exec();
-      returnData(plan, res);
+      let deletedPlan = {
+        ...plan._doc,
+        distributions: [],
+        fine_requirements: [],
+      };
+      returnData(deletedPlan, res);
     })
     .catch((err) => errorHandler(res, 400, err));
 });
 
 //***need to consider not allow user to change major for a plan ***/
+// updates a plan's major(s) and name
 router.patch("/api/plans/update", (req, res) => {
   const id = req.body.plan_id;
   const majors = req.body.majors;
+  const major_ids = req.body.major_ids;
   const name = req.body.name;
   if (!(majors || name)) {
     errorHandler(res, 400, "Must update majors or name.");
+  } else if ((majors && !major_ids) || (major_ids && !majors)) {
+    errorHandler(res, 400, "Must supply both major names and ids.");
   } else {
     let updateBody = {};
     if (majors) {
       updateBody.majors = majors;
+      updateBody.major_ids = major_ids;
     }
     if (name) {
       updateBody.name = name;
     }
     plans
       .findByIdAndUpdate(id, updateBody, { new: true, runValidators: true })
-      .then((plan) => {
-        reviews
+      .then(async (plan) => {
+        // add dists for new major, if any
+        await addMajorDistributions(plan);
+        // remove dists and fineReqs for deleted major, if any
+        await distributions.find({ plan_id: plan._id }).then(async (dists) => {
+          for (let dist of dists) {
+            if (!plan._doc.major_ids.includes(dist.major_id)) {
+              // maintain courses array fields
+              await courses.updateMany(
+                { plan_id: id },
+                { $pull: { distribution_ids: dist._id } }
+              );
+              for (let f_id of dist.fineReq_ids) {
+                await courses.updateMany(
+                  { plan_id: id },
+                  { $pull: { fineReq_ids: f_id } }
+                );
+              }
+              // delete documents
+              await distributions.findByIdAndDelete(dist._id);
+              await fineRequirements.deleteMany({ distribution_id: dist._id });
+            }
+          }
+        });
+        // return plan with reviews and distributions
+        let distObjs = await distributions.find({ plan_id: plan._id });
+        let fineObjs = await fineRequirements.find({ plan_id: plan._id });
+        await reviews
           .find({ plan_id: id })
           .populate("reviewer_id")
           .then((revs) => {
-            plan = { ...plan, reviewers: revs };
+            plan = {
+              ...plan,
+              distributions: distObjs,
+              fine_requirements: fineObjs,
+              reviewers: revs,
+            };
             returnData(plan, res);
```

Comment:
delete all of plan's distributions or just the added / deleted ones? 
it's not easy to determine whether we added or deleted major(s)